scripts/gmsa/domain.init.tmpl

```diff
@@ -0,0 +1,228 @@
+      path: C:\gmsa\create-AD.ps1
+      permissions: 0644
+    - content: |
+        Param(
+            [Parameter(Position = 0, Mandatory = $true)] [String] $AccountName,
+            [Parameter(Position = 1, Mandatory = $true)] [String] $pricipalId,
+            [Parameter(Position = 1, Mandatory = $true)] [String] $secretUri
+        )
+
+        # Logging for troubleshooting
+        Start-Transcript -Path "C:\gmsa\CredSpec.txt"
+        # exit on error
+        Set-StrictMode -Version Latest
+        $ErrorActionPreference = 'Stop'
+        $ProgressPreference = "SilentlyContinue"
+
+        # deps install
+        Install-PackageProvider -Name NuGet -Force
+        Install-Module powershell-yaml -Force
+        $ADDomain = Get-ADDomain -Current LocalComputer
+
+        # Clean up account names and validate formatting
+        $AccountName = $AccountName.TrimEnd('$')
+        $output = [ordered]	@{
+            "apiVersion" = "windows.k8s.io/v1"
+            "kind" = "GMSACredentialSpec"
+            "metadata" = @{
+                "name" = $AccountName
+            }
+            "credspec" = @{
+                "CmsPlugins" = @(
+                    "ActiveDirectory"
+                )
+                "DomainJoinConfig" = @{
+                    "Sid" = $ADDomain.DomainSID.Value
+                    "MachineAccountName" = $AccountName
+                    "Guid" = $ADDomain.ObjectGUID
+                    "DnsTreeName" = $ADDomain.Forest
+                    "DnsName" = $ADDomain.DNSRoot
+                    "NetBiosName" = $ADDomain.NetBIOSName
+                }
+                "ActiveDirectoryConfig" = @{
+                    "GroupManagedServiceAccounts" = @(
+                        @{
+                            "Name" = $AccountName
+                            "Scope" = $ADDomain.DNSRoot
+                        },
+                        @{
+                            "Name" = $AccountName
+                            "Scope" = $ADDomain.NetBIOSName
+                        }
+                    )
+                    "HostAccountConfig" = @{
+                        "PortableCcgVersion" = "1"
+                        "PluginGUID" = "{CCC2A336-D7F3-4818-A213-272B7924213E}"
+                        "PluginInput" = "ObjectId=$($pricipalId);SecretUri=$${secretUri}"
+                    }
+                }
+            }
+        }
+
+        $ManifestFile = "gmsa-cred-spec-$AccountName.yml"
+        $SpecPath = "C:\gmsa"
+        $ManifestFile = Join-Path $SpecPath "gmsa-cred-spec-gmsa-e2e.yml"
+        ConvertTo-Yaml $output | Set-Content $ManifestFile
+        Write-Output "K8S manifest rendered at $ManifestFile"
+      path: C:\gmsa\cred-spec.ps1
+      permissions: 0644
+users:
+    - groups: Administrators
+      name: capi
+      ssh_authorized_keys:
+        - ${AZURE_SSH_PUBLIC_KEY:=""}
+runcmd:
+    - schtasks.exe /create /tn "Domain Create" /ru SYSTEM /Sc ONSTART /tr "powershell /c C:/gmsa/create-AD.ps1"

```

Comment:
create-AD.ps1 should probably un-schedule the this scheduled task once setup is complete.

Reply:
Added `Disable-ScheduledTask -TaskName "Domain Create"` a few lines above